app/workers/work-item-update-queue-processor.ts

```diff
@@ -0,0 +1,47 @@
+import { Logger } from 'winston';
+import { batchProcessQueue } from '../backends/workflow-orchestration';
+import env from '../util/env';
+import { Worker } from './worker';
+import sleep from '../util/sleep';
+import { WorkItemUpdateQueueType } from '../util/queue/queue';
+
+export interface WorkItemUpdateQueueProcessorConfig {
+  logger: Logger;
+  queueType: WorkItemUpdateQueueType;
+}
+
+export default class WorkItemUpdateQueueProcessor implements Worker {
+  isRunning: boolean;
+
+  logger: Logger;
+
+  queueType: WorkItemUpdateQueueType;
+
+  constructor(config: WorkItemUpdateQueueProcessorConfig) {
+    this.logger = config.logger;
+    this.queueType = config.queueType;
+  }
+
+  async start(): Promise<void> {
+    this.isRunning = true;
+    let firstRun = true;
+    while (this.isRunning) {
+      if (!firstRun) {
+        await sleep(env.workItemUpdateQueueProcessorPeriodSec * 1000);
```

Comment:
Why do we have this sleep here? Shouldn't it start immediately because of the
```
WaitTimeSeconds: 20
```
in the queue configuration? Does this parameter need to be in sync with that one?

Reply:
Moved this sleep into the `catch` block to not sleep normally, but will sleep if there is an error to avoid a tight loop.